### ISSUE TITLE
creating a client_connection with terraform deletes preexisting client connections made in the gui

### ISSUE THREAD

**Issue by drewmullen** [appgate/terraform-provider-appgatesdp#123]
we had 5 pre-existing client connection definitions, mostly local but a few were type okta. all the ones not expressed in HCL were **deleted without warning**

plan output:
```
  # appgatesdp_client_connections.acme will be created
  + resource "appgatesdp_client_connections" "acme" {
      + id       = (known after apply)
      + spa_mode = "UDP-TCP"

      + profiles {
          + identity_provider_name = "acme"
          + name                   = "acme-okta"
          + spa_key_name           = "acme"
          + url                    = (known after apply)
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy
```

this was created but all pre-existing client-connections were **deleted**

**Comment by drewmullen** [comment]
something else I noticed:

sometimes when multiple client connections are created it deletes the others. i cant pin down a consistent example however, the behavior seems to change with each run. try defining multiple and then see how it behaves 1st run, 2nd run, then make updates to the values and it behaves differently again
```
resource "appgatesdp_client_connections" "acme" {
  spa_mode = "UDP-TCP"
  profiles {
    name                   = "acme-okta"
    spa_key_name           = "acme"
    identity_provider_name = "acme"
  }
}

resource "appgatesdp_client_connections" "dev" {
  spa_mode = "UDP-TCP"
  profiles {
    name                   = "dev"
    spa_key_name           = "dev"
    identity_provider_name = "local"
  }
}
```

perhaps the connections still exist and arent shown by the gui? Occassionally we get this error:
> │ Error: Could not update Client Connections 409 Conflict
│
│   with appgatesdp_client_connections.acme,
│   on identity_provider.tf line 93, in resource "appgatesdp_client_connections" "acme":
│   93: resource "appgatesdp_client_connections" "acme" {

**Comment by drewmullen** [comment]
another point supporting the theory that client connections still exist in the API (but are hidden by the gui), my local client user still has access to resources that came from the SPA key issued by the client connection (which doesnt appear to exist anymore). you'd think my access would fail or appgate client would error / force logoff

we attempted to refresh the client certs under `user & devices` menu, same result i still have access

**Comment by gokhan-berberoglu** [comment]
@drewmullen I want to clarify how Client Profile Links work:

the Profile Links are Appgate SDP's way of passing necessary configuration to Clients. They are not access control mechanisms, so deleting a profile link in the Controller *may* not prevent access from Clients that are configured that profile link. However, when you delete a profile link and *if* there are no other profile links with teh same spa key name they won't be able to connect anymore. 

Let me explain with an example:

lets say you have 3 profiles configured in the Controller with mutliple clients on each:

```json
[
  {"name": "dev1", "spa_key_name": "dev", "identity_provider_name": "local"},
  {"name": "dev2", "spa_key_name": "dev", "identity_provider_name": "local"},
  {"name": "prod", "spa_key_name": "prd", "identity_provider_name": "local"},
]
```
then you deleted "dev1":
```json
[
  {"name": "dev2", "spa_key_name": "dev", "identity_provider_name": "local"},
  {"name": "prod", "spa_key_name": "prd", "identity_provider_name": "local"},
]
```
all clients configured with the deleted dev1 will continue to work, becase they are using spa-key named "dev" and it still exist in the controller, because dev2 profile has a reference to it.

lets say then you deleted "dev2" also:
```json
[
  {"name": "prod", "spa_key_name": "prd", "identity_provider_name": "local"},
]
```
at this point, all clients configured with "dev1" & "dev2" profile links won't be able to connect, becuase they're using a spa-key named "dev" and that does not exist in the controller anymore. 

so connection depends on if the spa-key used by the client exist in the Controller or not.

I undertand this may be unintuitive, however there are valid reasons for the behaviour.

**one last note**: lets say you accidently deleted the "prod" profile link also in the Controller. In that case you may have thousands of users that cannot connect, as spa-key named "prd" does not exist anymore.  But the solution is pretty simple, you can create a new client profile link using the same spa-key name "prd" and all æffected clients will be able connect again. So spa-key content is tied to the spa-key name.

hope that helps :)

**Comment by drewmullen** [comment]
@gokhan-berberoglu that is fantastically helpful. thanks very much! :D

However, that only makes me more concerned that the provider is deleting pre-existing clients 😅  I also noticed that if you define 2+ client connection, terraform attempts to make changes every `apply`. I suspect that may by cause by a mismatch between the read APIs return, probably unordered, and the terraform provider expecting an [ordered list](https://github.com/appgate/terraform-provider-appgatesdp/blob/7704c44c61c75a96e2eb355e79fb04c829746458/appgate/resource_appgate_client_connections.go#L40)

**Comment by danijeel** [comment]
We have moved client profile to its own resource,  `appgatesdp_client_profile` and deprecated `appgatesdp_client_connections`

Example usage
```hcl

resource "appgatesdp_client_profile" "client_profile" {
  name                   = "development"
  spa_key_name           = "development"
  identity_provider_name = "local"
}


```

this is included in [0.5.6](https://github.com/appgate/terraform-provider-appgatesdp/releases/tag/v0.5.6)

**Comment by drewmullen** [comment]
thanks! ill try this out this week 👍
